rapids-cmake/cython/add_rpath_entries.cmake

```diff
@@ -72,14 +72,19 @@ function(rapids_cython_add_rpath_entries)
     list(APPEND cleaned_paths "${path}")
   endforeach()
 
+  if (CMAKE_SYSTEM_NAME STREQUAL "Linux")
```

Comment:
Could we make this `@loader_path` behavior specific to Darwin and use `$ORIGIN` in an "else"? I'd like to avoid the string check for `"Linux"` as a defensive behavior.

Reply:
I agree, we should explicit check for Darwin and otherwise use $ORIGIN